components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuCandidate.kt

```diff
@@ -262,6 +262,8 @@ private fun HitResult.isLink(): Boolean =
 internal fun HitResult.getLink(): String = when (this) {
     is HitResult.UNKNOWN -> src
     is HitResult.IMAGE_SRC -> uri
+    is HitResult.IMAGE ->
+        if (title.isNullOrBlank() || title?.startsWith("http", true) == true) src else title.toString()
```

Comment:
`|| title?.startsWith("http", true) == true` I don't think we need this part right? If the title is null or blank that should be enough.

Reply:
There was a discussion on https://github.com/mozilla-mobile/fenix/issues/580 if we should consider not showing titles that could hide the real src of the image. If you think it is not necessary, we can change that.

Reply:
I did not add tests for the "http" issue. Please let me know if I should change the getLink code or add a test.

Reply:
Yeah, `title?.startsWith("http", true)` looks fishy. This is true for titles that are not URLs.

Reply:
The concern was to not allow images to hide real url by using a title.
https://www.kevinbrosnan.net/testcases/absurdly-long-title-text.html
If you test it on Chrome, you can see they they also exchange title for src when the title starts with "http"

Reply:
> The concern was to not allow images to hide real url by using a title.
> https://www.kevinbrosnan.net/testcases/absurdly-long-title-text.html
> If you test it on Chrome, you can see that they also exchange title for src when the title starts with "http"

Would `title.isUrl()` work in that case?

Reply:
For what it's worth, the test page spoofing doesn't work on r-b so I'm not sure if this is needed.

Reply:
I guess we could use another way to check if it is an url or it poses as one. I haven't checked the code on reference browser, but in fenix, without that check it shows the title with http.